src/vm/analysis/type_checker/mod.rs

```diff
@@ -367,6 +376,9 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
             },
             List(ref expression) => {
                 self.type_check_function_application(expression, context)?
+            },
+            TraitReference(_) | Field(_) => {
+                return Err(CheckErrors::InvalidTypeDescription.into());
```

Comment:
This should probably be something like a `MisplacedTraitOrFieldReference` error, since this occurs if a TraitReference (`<trait>`) or field is used anywhere other than a `use-trait` or define argument, right?

Reply:
I initially wanted to reuse existing error types, but introducing a new makes sense.
Addressed with a198bd1, which also add a test case.